### PR TITLE
`invocations`: Type Annotations

### PR DESCRIPTION
**Pull request by cumbof**
Type annotating `invocations`.

Validating with: `tox -e lint`
Still working on this. Here is the result of the linting:

```bash
lint run-test: commands[0] | flake8 .
lint run-test: commands[1] | black --check --diff .
All done! ✨ 🍰 ✨
114 files would be left unchanged.
lint run-test: commands[2] | isort --check --diff .
Skipped 3 files
lint run-test: commands[3] | mypy bioblend/
bioblend/galaxy/invocations/__init__.py:442: error: "GalaxyClient" has no attribute "invocations"  [attr-defined]
                invocation = self.gi.invocations.show_invocation(invocation_id)
                             ^
Found 1 error in 1 file (checked 67 source files)
ERROR: InvocationError for command /Users/fabio/GitHub/bioblend/.tox/lint/bin/mypy bioblend/ (exited with code 1)
__________________________________________________________________ summary ___________________________________________________________________
ERROR:   lint: commands failed
```

### CODE REVIEW

**Comment by cumbof** [comment]
@nsoranzo

Line `442`:
```python
invocation = self.gi.invocations.show_invocation(invocation_id)
```

Shouldn't be as following?
```python
invocation = self.show_invocation(invocation_id)
```

**Comment by cumbof** [comment]
Fixed by explicitly declaring `gi: "GalaxyInstance"`

File: bioblend/galaxy/invocations/__init__.py [hunk 1]
```diff
@@ -3,36 +3,46 @@
...
-        step_details=False,
-    ):
+        workflow_id: str = None,
+        history_id: str = None,
+        user_id: str = None,
+        include_terminal: bool = True,
+        limit: int = None,
+        view: str = "collection",
+        step_details: bool = False,
+    ) -> List[Dict[str, str]]:
```
Comment (nsoranzo) [inline bioblend/galaxy/invocations/__init__.py:45]:
```suggestion
    ) -> List[Dict[str, Any]]:
```

File: bioblend/galaxy/invocations/__init__.py [hunk 2]
```diff
@@ -401,7 +411,9 @@ def get_invocation_biocompute_object(self, invocation_id):
         url = self._make_url(invocation_id) + "/biocompute"
         return self._get(url=url)
 
-    def wait_for_invocation(self, invocation_id, maxwait=12000, interval=3, check=True):
+    def wait_for_invocation(
+        self, invocation_id: str, maxwait: int = 12000, interval: int = 3, check: bool = True
```
Comment (nsoranzo) [inline bioblend/galaxy/invocations/__init__.py:415]:
```suggestion
        self, invocation_id: str, maxwait: float = 12000, interval: float = 3, check: bool = True
```

File: bioblend/galaxy/invocations/__init__.py [hunk 3]
```diff
@@ -443,7 +455,7 @@ def wait_for_invocation(self, invocation_id, maxwait=12000, interval=3, check=Tr
                     f"Invocation {invocation_id} is still in non-terminal state {state} after {maxwait} s"
                 )
 
-    def _invocation_step_url(self, invocation_id, step_id):
+    def _invocation_step_url(self, invocation_id, step_id: str) -> str:
```
Comment (nsoranzo) [inline bioblend/galaxy/invocations/__init__.py:458]:
```suggestion
    def _invocation_step_url(self, invocation_id: str, step_id: str) -> str:
```

Comment (cumbof) [reply bioblend/galaxy/invocations/__init__.py:458]: Oh, I missed the `invocation_id`.. Thanks!

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
Thanks @cumbof !

**Pull request merged**